src/pages/SmartSwatch.js

```diff
@@ -18,54 +20,114 @@ import {
   Text,
   Link,
   useColorMode,
+  useClipboard,
 } from '@chakra-ui/core';
 import { FaPalette, FaHeart } from "react-icons/fa";
 
 import { Swatch } from '../components/Swatch';
 import { Logo } from '../components/Logo';
 
+const PRESET_COLORS = [
+  '#C70833',
+  '#E53E3E',
+  '#F6AD55',
+  '#F6E05E',
+  '#48BB78',
+  '#4FD1C5',
+  '#4299E1',
+  '#0BC5EA',
+  '#805AD5',
+  '#D53F8C',
+  '#718096',
+];
+const DEFAULT_USER_COLOR_STRING = PRESET_COLORS[0];
+const UNKNOWN_USER_COLOR_STRING = '#000';
+const LIGHTNESS_MAP = [0.95, 0.85, 0.75, 0.65, 0.55, 0.45, 0.35, 0.25, 0.15, 0.05];
+const SATURATION_MAP = [0.32, 0.16, 0.08, 0.04, 0, 0, 0.04, 0.08, 0.16, 0.32];
+const HUE_MAP = [0, 4, 8, 12, 16, 20, 24, 28, 32, 36];
+
+const getColorStringFromHash = () => {
+  const sanitizedHash = window.location.hash.replace('#', '');
+  return chroma.valid(sanitizedHash) ? chroma(sanitizedHash).name() : DEFAULT_USER_COLOR_STRING;
+}
+
+const getUserColorChroma = (colorString, fallbackValue = UNKNOWN_USER_COLOR_STRING) => chroma.valid(colorString)
+  ? chroma(colorString)
+  : chroma(fallbackValue);
+
 export const SmartSwatch = () => {
-  const [userColorInput, setUserColorInput] = useState('#C70833');
+  const [userColorInput, setUserColorInput] = useState(getColorStringFromHash());
+  const [userColorChroma, setUserColorChroma] = useState(getUserColorChroma(userColorInput.trim()));
   const { colorMode, toggleColorMode } = useColorMode();
 
-  const lightnessMap = [0.95, 0.85, 0.75, 0.65, 0.55, 0.45, 0.35, 0.25, 0.15, 0.05];
-  const saturationMap = [0.32, 0.16, 0.08, 0.04, 0, 0, 0.04, 0.08, 0.16, 0.32];
-  const hueMap = [0, 4, 8, 12, 16, 20, 24, 28, 32, 36];
+  const { onCopy, hasCopied } = useClipboard(window.location);
+
+  React.useEffect(function updateColorChromaFromInputEffect() {
```

Comment:
you can use anonymous arrow function in useEffect 😉
```
React.useEffect(() => {
```

Reply:
Yeah I know! I'm just using named functions rather than anonymous arrow ones as good practice to 1) make easier to understand what's the purpose of the effect and 2) make easier to spot the effect in react dev tools for debugging.

![CleanShot 2020-05-25 at 18 14 34](https://user-images.githubusercontent.com/2677072/82829472-a4d55380-9eb3-11ea-8a40-cf23a03a2dea.png)

Reply:
Oh ok, nice :) You can keep them!

Reply:
Thanks for the tips :)